addOns/client/src/main/java/org/zaproxy/addon/client/ExtensionClientIntegration.java

```diff
@@ -107,6 +109,8 @@ public class ExtensionClientIntegration extends ExtensionAdaptor {
     private ClientSpiderDialog spiderDialog;
     private ZapMenuItem menuItemCustomScan;
 
+    private List<AuthenticationHandler> authHandlers = new ArrayList<>();
```

Comment:
Unlikely but this should take into account concurrency.

Reply:
Could you be more specific?

Reply:
The spider iterates the list and the list can be modified at the same time.

Reply:
Sorry, I meant, what changes/fixes are needed. Just a synchronized list?